### ISSUE TITLE
[BUG] volumes do not mount in the pods

### ISSUE THREAD

**Issue by IJOL** [longhorn/longhorn#1947]
I keep getting this messages from my pods, 

Unable to attach or mount volumes: 

When this was after a resizing by the manual, so i scaled deployments using volumes to 0, modified PVC with new size, but after scaling de deployments, get stuck with this message related to longhorn volumes..

Which are the logs relevant to that problems? manager, engine  or csi-attacher.. i try to fnd somethign relevant but to no avail, 

Any help will be reatly aprecitted we are in a hurry, we tried too tio recreate pvc and pv , from backups  but they stuck at attaching/mounting..,.

**Comment by IJOL** [comment]
Which is the process that controls attachement/mounting? how can i restart it, if it has sense, and is usefull?

**Comment by IJOL** [comment]
It's responding but i delayed for 2 hours more or less.  the bug an the problem remains we can not wiat for 2 to 3 hours in mounting a volyume... what can be done.. we coninue in a hurry

**Comment by IJOL** [comment]
LOL everything is going back to normal after hours, it's amazing, i think it will be the same as in #1154 but in the next level, we are experiencing hours delay attaching and reattaching and after 1 hour or more mounting,

**Comment by yasker** [comment]
@IJOL Can you send us a support bundle? Any attach/detach should be in the matters of seconds.

Are you use `fsGroup`? Or it sounds like the resizing might be incomplete when you scale back the workload, so it's waiting on it.

**Comment by IJOL** [comment]
How can i sen you the support bundle? now the cluster seems stable, the workloads get volumes but it's painfully slow, for anewly created volume, not a resized one, we see abot 15-30 minutes attaching/mounting, we ca save some time if we attach volumes ourselves with the UI,

**Comment by yasker** [comment]
You can send the support bundle to longhorn-support-bundle@rancher.com .

**Comment by IJOL** [comment]
Sending the support bundle to the email provided but, from comments in slack by Josuha Moody that points to https://github.com/longhorn/longhorn/issues/1463#issuecomment-664679380 , it seems to me a duplicate of this other ones.. we will try to upgrade our cluster to a newer 1.18.x version ..

**Comment by IJOL** [comment]
Problem solved for the record, finally we simply updated our rke generated cluster to k8s v 1.18.10, upgraded related components, for the gory details, we upgraded the rke tool itself, to v1.1.11 latest stable, rke up, and everything was fine to get to know which k8s version a given release of rke upgrades and supports, rke config -l -a

**Comment by yasker** [comment]
Great to hear it's solved @IJOL . For the record, the support bundle shows 75 volumes on 4 nodes. It's less than 200 volumes which is the limit last time we saw in #1463. But we don't collect kubelet log in the support bundle so I cannot say if it's the same issue or not. Feel free to reopen the issue if it happened again.

**Comment by IJOL** [comment]
Just as an aside, which kind of machines (RAM,CPU) and how many nodes were active in that other case #1463 ? for the sake of comparison and be prepared  for problems maybe coming to us, we are absolute rookies in regard to k8s, but we are good readers..:)

**Comment by joshimoo** [comment]
@IJOL #1463 has to do with the kubernetes api server and is a result of an inefficient csi volume lookup. So it's possible to reach that condition with less than 200 csi volumes in the case where you might be running an under provisioned control plane or general higher IO times controler server <-> api server <-> etcd

The inefficient csi lookup implementation got fixed in kubernetes version 1.18.5 / 1.17.8
Below are the kubernetes recommended control plane requirements for different cluster sizes.
https://kubernetes.io/docs/setup/best-practices/cluster-large/#size-of-master-and-master-components

**Comment by IJOL** [comment]
@joshimoo After read that #1463  was triggered with about 200 volumes and ours appeared at 72 vol, we are trying to double check what we are doing, in respect to performance of the k8s cluster, we have a combined 3 nodes all etcd, controlplane and worker at the same time, we know this i not recommended at all.. hence our little fear that maybe we are a our practical maximun.. with just about 72 volumes,,
